scipy/stats/tests/test_distributions.py

```diff
@@ -3683,31 +3710,25 @@ def test_ppf(self, p, a, b, expected):
         assert_allclose(x, expected, rtol=1e-14)
 
     @pytest.mark.parametrize(
-        'x, a, b, expected',
-        [(1e22, 0.05, 0.1, 0.9978811466052919),
-         (1e22, 100.0, 0.05, 0.8973027435427167),
-         (1e10, 0.05, 0.1, 0.9664184367890859),
-         (1e10, 1.5, 1.5, 0.9999999999999983),
-         (1e10, 100.0, 0.05, 0.5911548582766262),
-         (1e-10, 0.05, 0.1, 0.21238845427095),
-         (1e-10, 1.5, 1.5, 1.697652726007973e-15),
-         (1e-10, 0.05, 100.0, 0.40884514172337383),
-         (1e-22, 0.05, 0.1, 0.053349567649287326),
-         (1e-22, 1.5, 1.5, 1.6976527263135503e-33),
-         (1e-22, 0.05, 100.0, 0.10269725645728331),
-         (1e-100, 0.05, 0.1, 6.7163126421919795e-06),
-         (1e-100, 1.5, 1.5, 1.6976527263135503e-150),
-         (1e-100, 0.05, 100.0, 1.2928818587561651e-05)])
-    def test_cdf_tails(self, x, a, b, expected):
-        # test for gh-17631
-        # values computed with mpmath.
-        # Example:
-        # from mpmath import mp
-        # mp.dps = 50
-        # a, b = mp.mpf(0.05), mp.mpf(0.1)
-        # x = mp.mpf(1e22)
-        # float(mp.betainc(a, b, 0.0, x/(1+x), regularized=True))
-        assert_allclose(stats.betaprime.cdf(x, a, b), expected, rtol=1e-14)
+        'x, a, b, p, tol',
+        cdf_vals + [
+            (1e22, 0.05, 0.1, 0.9978811466052919, 1e-12),
+            (1e22, 10, 0.01, 0.38019826239901977, 1e-13),
+            (1e8, 1.5, 1.5, 0.9999999999983024, 1e-5),
+            (1e50, 0.05, 0.1, 0.9999966641709545, 1e-1),
+            (1e50, 100.0, 0.05, 0.995925162631006, 1e-1),]
```

Comment:
yeah, you are right, one has to be careful when reusing the cdf values to test the ppf

i have simplified it a bit, using only rtol=1e-14

I dropped a few test cases since we already have good coverage of different combinations

even for the extreme case x=1e50, the error is much smaller if one uses Warren's approch to compute the ppf target value. i decided to drop these cases, we consider x=1e22, that is already quite extreme.